include/eve/arch/abi_of.hpp

```diff
@@ -7,85 +7,84 @@
+template<typename Type, int Cardinal> struct abi_of
+{
+  static constexpr auto find()
   {
-    static constexpr auto find()
+    if constexpr( supports_simd )
     {
-      if constexpr(supports_simd)
-      {
-        constexpr auto width  = sizeof(Type) * Cardinal;
-        constexpr bool f64    = std::is_same_v<Type,double>;
+      constexpr auto width = sizeof(Type) * Cardinal;
+      constexpr bool f64   = std::is_same_v<Type, double>;
 
-        if constexpr( spy::simd_instruction_set == spy::x86_simd_ )
-        {
-               if constexpr( width <= 16) return x86_128_{};
-          else if constexpr( width == 32) return x86_256_{};
-          else if constexpr( width == 64) return x86_512_{};
-          else                            return aggregated_{};
-        }
-        else if constexpr( spy::simd_instruction_set >= spy::vsx_ )
-        {
-          if constexpr(width <= 16) return ppc_{};
-          else                      return emulated_{};
-        }
-        else if constexpr( spy::simd_instruction_set >= spy::vmx_ )
-        {
-          if constexpr(!f64 && width <= 16) return ppc_{};
-          else                              return emulated_{};
-        }
-        else if constexpr( spy::simd_instruction_set == spy::fixed_sve_ )
-        {
-          if constexpr(spy::simd_instruction_set.width == 128)      return arm_sve_128_{};
-          else if constexpr(spy::simd_instruction_set.width == 256) return arm_sve_256_{};
-          else if constexpr(spy::simd_instruction_set.width == 512) return arm_sve_512_{};
-          else                                                      return emulated_{};
-        }
-        else if constexpr( spy::simd_instruction_set == spy::arm_simd_ )
+      if constexpr( spy::simd_instruction_set == spy::wasm_simd_ )
+      {
+        if constexpr( width <= 16 ) return wasm_128_ {};
+        else return aggregated_ {};
+      }
+      else if constexpr( spy::simd_instruction_set == spy::x86_simd_ )
+      {
+        if constexpr( width <= 16 ) return x86_128_ {};
+        else if constexpr( width == 32 ) return x86_256_ {};
+        else if constexpr( width == 64 ) return x86_512_ {};
+        else return aggregated_ {};
+      }
+      else if constexpr( spy::simd_instruction_set >= spy::vsx_ )
+      {
+        if constexpr( width <= 16 ) return ppc_ {};
+        else return emulated_ {};
+      }
+      else if constexpr( spy::simd_instruction_set >= spy::vmx_ )
+      {
+        if constexpr( !f64 && width <= 16 ) return ppc_ {};
+        else return emulated_ {};
+      }
+      else if constexpr( spy::simd_instruction_set == spy::fixed_sve_ )
+      {
+        if constexpr( spy::simd_instruction_set.width == 128 ) return arm_sve_128_ {};
+        else if constexpr( spy::simd_instruction_set.width == 256 ) return arm_sve_256_ {};
+        else if constexpr( spy::simd_instruction_set.width == 512 ) return arm_sve_512_ {};
+        else return emulated_ {};
```

Comment:
Ok but if we are not supporting bigger sve for now, shouldn't we go to neon here?